sql_grader/mixins/grading.py

```diff
@@ -1,21 +1,72 @@
 """
 Mixin grading-related functionality
 """
+import logging
+import os.path
+from xblock.fields import Boolean
 from xblock.fields import Float
 from xblock.fields import Integer
 from xblock.fields import Scope
+from xblock.fields import String
+
 from xblock.scorable import ScorableXBlockMixin
 from xblock.scorable import Score
 
-from ..models import SqlProblem
-from ..models import DATABASES
+from ..models import all_datasets
+
+try:
+    from django.utils.translation import ugettext_lazy as _
+except ImportError:
+    def _(text):
+        """
+        Mock translation for scraping
+        """
+        return text
 
+log = logging.getLogger('sql_grader')
 
-def _(text):
+
+# pylint: disable=unused-argument
+def attempt_safe(dataset, answer_query, verify_query, is_ordered, query):
     """
-    Mock translation for scraping
+    Attempt a SqlProblem, using codejail to sandbox the execution.
     """
-    return text
+    # pylint: disable=E0401,C0415
+    from capa.safe_exec import safe_exec
+    from codejail.safe_exec import SafeExecException
+    results = {}
+    code = '''
+from sql_grader.models import SqlProblem
+submission_result, answer_result, error, comparison = SqlProblem(
+    answer_query={answer_query!r},
+    dataset={dataset!r},
+    verify_query={verify_query!r},
+    is_ordered={is_ordered}
+).attempt({query!r})
+
+'''.format(**locals())
+    python_path = [os.path.abspath(
+        os.path.join(
+            os.path.dirname(__file__),
+            '../..'
+        )
+    )]
+
+    try:
+        safe_exec(code, results, python_path=python_path, slug='sql_grader')
+    except SafeExecException:
+        log.exception(query)
+        # how should resource limits be communicated to the user?
```

Comment:
@stvstnfrd is the closest to the course team, but I think it's fine to leave it generic.